File: src/main/Java/boj_step16/boj28278.java
import java.io.*;
import java.util.Stack;
import java.util.StringTokenizer;

public class boj28278 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(reader.readLine());
        Stack<Integer> stack = new Stack<>();
        while (n-- > 0) {
            StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
            switch (tokenizer.nextToken()) {
                case "1": stack.push(Integer.parseInt(tokenizer.nextToken())); break;
                case "2": {
                    if (!stack.isEmpty()) writer.append(stack.pop() + "\n");
                    else writer.append("-1\n");
                } break;
                case "3": writer.append(stack.size() + "\n"); break;
                case "4": writer.append((stack.isEmpty()) ? "1\n" : "0\n"); break;
                case "5": {
                    if (!stack.isEmpty()) writer.append(stack.lastElement() + "\n");
                    else writer.append("-1\n");
                }
            }
        }
        writer.close();
    }
}

File: src/main/kotlin/boj_step16/boj28278.kt
import java.io.BufferedWriter
import java.io.OutputStreamWriter
import java.util.*

fun main() {
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val r = Scanner(System.`in`)
    val n = r.nextInt()
    val stack = Stack<Int>()
    repeat(n) {
        val x = r.nextInt()
        if (x == 1) {
            val y = r.nextInt()
            stack.add(y)
        }
        if (x == 2) {
            if (stack.isNotEmpty()) {
                w.write(stack.pop().toString())
                w.newLine()
            } else w.write("-1\n")
        }
        if (x == 3) {
            w.write(stack.size.toString())
            w.newLine()
        }
        if (x == 4) {
            w.write(stack.isEmpty().compareTo(false).toString())
            w.newLine()
        }
        if (x == 5) {
            if (stack.isNotEmpty()) {
                w.write(stack.last().toString())
                w.newLine()
            } else w.write("-1\n")
        }
    }
    w.close()
}